### PR TITLE
Bugs/SignIn/CLK32 fixbug SignIn

### PR DESCRIPTION
**Pull request by Minhdn1**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // Explain what you have done

### Solution
- // Explain how you fix it

### Impact
- // List of affected components/modules

### Screenshots (Optional)
- // Capture affected screens

### CODE REVIEW

File: ClearKeep/Scenes/Home/Views/Server/JoinServerView.swift [hunk 1]
```diff
@@ -21,7 +21,8 @@ struct JoinServerView: View {
 	@State private var serverURLInputStyle: TextInputStyle = .default
 	@State private var isValid: Bool = false
 	@State private var navigateToLogin: Bool = false
-	
+	@State private(set) var customServer: CustomServer = CustomServer()
+//	@State private var navigateToHome: Bool = false
```
Comment (vmo-klinh) [inline ClearKeep/Scenes/Home/Views/Server/JoinServerView.swift]: remove unused code

File: ClearKeep/App/Resources/Localize.strings [hunk 2]
```diff
@@ -86,6 +86,11 @@
 "Login.ForgotPassword" = "Forgot Password?";
 "Login.SignUp.Suggest" = "Don't have an account?";
 "Login.SignUp" = "Sign up";
+"Login.Popup.EmailValid" = "Email is invalid";
+"Login.Popup.Email.password.Validate" = "Email or Password is incorrect";
```
Comment (vmo-klinh) [inline ClearKeep/App/Resources/Localize.strings]: .password not uppercase

File: ClearKeep/Scenes/Login/Models/LoginViewPopUp.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+//
+
+import Foundation
+import Networking
+
+enum LoginViewPopUp {
+
+	case invalid
+	case invalidEmail
+	case emailblank
```
Comment (vmo-klinh) [inline ClearKeep/Scenes/Login/Models/LoginViewPopUp.swift]: case-sensitive, should be emailBlank

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 4]
```diff
@@ -69,9 +71,30 @@ struct LoginContentView: View {
...
+			switch activeAlert {
+			case .invalid:
+				return Alert(title: Text(activeAlert.title),
+							 message: Text(activeAlert.message),
+							 dismissButton: .default(Text(activeAlert.primaryButtonTitle)))
+			case .invalidEmail:
+				return Alert(title: Text(activeAlert.title),
+							 message: Text(activeAlert.message),
+							 dismissButton: .default(Text(activeAlert.primaryButtonTitle)))
+			case .emailblank:
```
Comment (vmo-klinh) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: case-sensitive

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 5]
```diff
@@ -168,10 +185,26 @@ private extension LoginContentView {
...
+		let emailValidate = injected.interactors.loginInteractor.emailValid(email: email)
+		emailValidate ? doLogin() : ({ self.activeAlert = .invalid })()
+		self.isShowAlertLogin = true
+	}
+	
 	func doLogin() {
 		loadable = .isLoading(last: nil, cancelBag: CancelBag())
 		Task {
-			loadable = await injected.interactors.loginInteractor.signIn(email: email, password: password, customServer: customServer)
+			loadable = await injected.interactors.loginInteractor.signIn(email: email.trimmingCharacters(in: .whitespaces), password: password, customServer: customServer)
```
Comment (vmo-klinh) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: should use .whitespacesAndNewlines

File: ClearKeep/Scenes/Home/Views/Server/JoinServerView.swift (hunk 1 above)
Comment (Minhdn1) [reply ClearKeep/Scenes/Home/Views/Server/JoinServerView.swift]: fix complete

File: ClearKeep/App/Resources/Localize.strings (hunk 2 above)
Comment (Minhdn1) [reply ClearKeep/App/Resources/Localize.strings]: fixcomplete

File: ClearKeep/Scenes/Login/Models/LoginViewPopUp.swift (hunk 3 above)
Comment (Minhdn1) [reply ClearKeep/Scenes/Login/Models/LoginViewPopUp.swift]: fix complete

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift (hunk 4 above)
Comment (Minhdn1) [reply ClearKeep/Scenes/Login/Views/LoginContentView.swift]: fix complete

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift (hunk 5 above)
Comment (Minhdn1) [reply ClearKeep/Scenes/Login/Views/LoginContentView.swift]: fix complete

**Review by vmo-klinh — APPROVED** [review]

**Pull request merged**